src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py

```diff
@@ -0,0 +1,230 @@
+                    blob_name=blobName,
+                    start_range=start,
+                    end_range=end,
+                    stream=stream)
+
+                currBytes = stream.getvalue()
+                newByteSize = len(currBytes)
+                amountRead = newByteSize - oldByteSize
+                start += amountRead
+                end = start + byteSize - 1
+
+                # Only scan what's newly read. If nothing is read,
+                # default to 0.
+                minScanRange = max(newByteSize - amountRead - 1, 0)
+                for i in range(newByteSize - 1, minScanRange, -1):
+                    if currBytes[i-1:i+1] == b'\r\n':
+                        flush = currBytes[:i]  # won't print \n
+                        stream = BytesIO()
+                        stream.write(currBytes[i+1:])
+                        print(flush.decode('utf-8'))
+                        break
+
+            except AzureHttpError as ae:
+                if ae.status_code != 404:
+                    sys.exit(ae)
+            except KeyboardInterrupt:
+                currBytes = stream.getvalue()
+                if len(currBytes) > 0:
+                    print(currBytes.decode('utf-8'))
+                sys.exit()
+            except:
+                raise
+
+        try:
+            props = blobService.get_blob_properties(
+                container_name=containerName, blob_name=blobName)
+            metadata = props.metadata
+            available = props.properties.content_length
+            lastModified = props.properties.last_modified
+
+        except AzureHttpError as ae:
+            if ae.status_code != 404:
+                sys.exit(ae)
+        except KeyboardInterrupt:
+            if len(currBytes) > 0:
+                print(currBytes.decode('utf-8'))
+            sys.exit()
+        except:
+            raise
+
+        # If we're still expecting data and we have a record for the last
+        # modified date and the last modified date has timed out, exit
+        if ((lastModified is not None and blob_is_not_complete(metadata)) or
+                start < available):
+
+            delta = datetime.utcnow().replace(tzinfo=pytz.utc) - lastModified
+
+            if delta.seconds > timeoutInSeconds:
+                print("No additional logs found. Timing out...")
+                sys.exit()
+
+        # If no new data available but not complete, sleep before trying
+        # to process additional data.
+        if (blob_is_not_complete(metadata) and start >= available):
+            time.sleep(5)
+
+
+def blob_is_not_complete(metadata):
+    if metadata is None:
+        return True
+
+    for key in metadata:
+        if key.lower() == 'complete':
+            return False
+
+    return True
+
+
+def acr_queue(cmd, client, registry_name, resource_group_name, source_location, docker_file_path, image_name, context_path=None, timeout=None, arguments=None, secret_arguments=None):
```

Comment:
I had concern with `image_name` even in swagger but let's keep it as it for now.